src/components/search-bar/search-result-header/SearchResultHeader.tsx

```diff
@@ -1,32 +1,67 @@
 import * as React from "react";
-import { useDispatch } from "react-redux";
-import { openModal } from "../../../core/modal.slice";
+
+import {
+  FilterItemTerm,
+  TermOnClickHandler
+} from "../../../apps/search-result/types";
+import {
+  FacetResult,
+  useSearchFacetQuery
+} from "../../../core/dbc-gateway/generated/graphql";
 import { useText } from "../../../core/utils/text";
-import { FacetBrowserModalId } from "../../facet-browser/FacetBrowserModal";
-import ButtonTag from "../../Buttons/ButtonTag";
+import {
+  formatFacetTerms,
+  getPlaceHolderFacets,
+  lineFacets
+} from "../../facet-browser/helper";
+import FacetLine from "../../facet-line/FacetLine";
+import FacetLineSelected from "../../facet-line/FacetLineSelected";
 
 export interface SearchResultHeaderProps {
   hitcount: string;
   q: string;
+  filters: { [key: string]: { [key: string]: FilterItemTerm } };
+  filterHandler: TermOnClickHandler;
 }
 
 const SearchResultHeader: React.FC<SearchResultHeaderProps> = ({
   hitcount,
-  q
+  q,
+  filters,
+  filterHandler
 }) => {
   const t = useText();
-  const dispatch = useDispatch();
+
+  const { data } = useSearchFacetQuery(
+    {
+      q: { all: q },
+      facets: lineFacets,
+      facetLimit: 10,
+      filters: formatFacetTerms(filters)
+    },
+    {
+      keepPreviousData: true,
+      placeholderData: {
+        search: {
+          facets: getPlaceHolderFacets(lineFacets)
+        }
+      }
+    }
+  );
 
   return (
     <>
       <h1 className="text-header-h2 mb-16 search-result-title">
         {`${t("showingResultsForText")} “${q}” (${hitcount})`}
       </h1>
-      <ButtonTag
-        onClick={() => dispatch(openModal({ modalId: FacetBrowserModalId }))}
-      >
-        {t("addMoreFiltersText")}
-      </ButtonTag>
+      {data && (
+        <FacetLine
+          filters={filters}
+          facets={data.search.facets as FacetResult[]}
+          filterHandler={filterHandler}
+        />
+      )}
+      <FacetLineSelected filters={filters} filterHandler={filterHandler} />
```

Comment:
Consider moving all facet components (the selectable terms and the selected terms) into a single element and then using this element as a part of the search result directly instead of passing it through the header.

This is the same approach as we use for the facet browser and the header is not really related to the facetline either.